### ISSUE TITLE
Deviating width axis

### ISSUE THREAD

**Issue by clauseggers** [clauseggers/Playfair#41]
Hi @emmamarichal With Playfair 2.1 done and Glyphs unable to fully generate working fonts, I wanted to try out `fontmake`. This came with a number of other issues, the mapping/location of axes but one of them. So I pulled up the version of Playfair you have in your fork, and compared it with the one I have from Glyphs. It turns out that the optical and weight axes match fairly close (probably kerning differences), but the width axis is quite different.

https://github.com/clauseggers/Playfair/assets/868987/15539c5b-a32a-4de6-9e61-b5f072bf096f

I am guessing that Glyphs is using all my defined static instances to create a compile-time axis mapping? Do you know?

Since Glyphs can’t generate working variable fonts, I’m pondering the idea of moving to using `fontmake` and I had a look at it yesterday. But there were many other issues, and many changes needs to be made for this to work. So here is the question: Can you compile a list or outline of things that need to be changed for the Glyphs files to be successfully generated in `fontmake`? I need first to find out if this manoeuvre is practically feasible.

**Comment by emmamarichal** [comment]
Hello @clauseggers!

I think the difference in the width axis is due of a different axis mapping (in my version, the axis starts from 83, but it doesn't affect the design). But otherwise, I tried to export it with gftools, and it works well, it just takes a lot of time to export all the statics. On my side, I use gftools, here is the command:

`gftools builder path/to/your/font`

> I am guessing that Glyphs is using all my defined static instances to create a compile-time axis mapping? Do you know?
Your defined statics will be used in the FVAR table, and will appear in dropdown menus.

If you want to keep your instances names, we are good, tell me if you notice something else that look weird or doesn't work. 
But, for the Google fonts version, I'll need to do the same process as before, with renaming instances, adding the stat table etc. But in that case, I'll do it, so you don't have to bother with this!

Here is the variable font I managed to export (only Roman styles): 
[variable.zip](https://github.com/clauseggers/Playfair/files/11623716/variable.zip)

**Comment by clauseggers** [comment]
Hi Emma
Will users have access to instance names in GF? If not, then they are not relevant on your side.

There _is_ an issue with different axes mappings. When users specify a ‘font stretch’ value, then they will get a different rendering depending on whether they are using my fonts, or the GF version.

I don’t know why you generated and attached the fonts, I already have them from your branch. I used them in the video I attached.

What I would like to have is – as I mentioned – an outline of which ways of doing things in the Glyphs file, will allow your system to generate the VFs without modification. That would include:
1) How to specify axes
2) Not having curves overlapping themselves in bracket layers
3) How to setup bracketed layers with nested components
&c, &c.

Those kinds of things.

**Comment by emmamarichal** [comment]
@clauseggers 
Ok I see! 
- First, instances names must follow our specs, and be named how I did in my file version. Concerning the axis mapping, I'll take a look to the problem you are mentioning, when the files will be ready.
- Indeed, some paths have to be merged in bracket layers, I mentioned it here : https://github.com/clauseggers/Playfair/pull/38#issuecomment-1450353812
- Concerning nested components in bracket layers, I see one case: tshe-cy, and I didn't have any trouble to export it last time, do you have some?

As I remember, I didn't meet a lot of design issues when I worked on it a few months ago, except merging some paths in bracket layers!

**Comment by clauseggers** [comment]
Alright – thank you Emma. The Google Fonts documentation is helpful. I’ll see how far I get on this info.

**Comment by clauseggers** [comment]
I had a look at the width axis issue, and the issue stems from each of us remapping the new ‘100’ value to a different coordinate. I have mapped 75→100, whereas you have mapped 100→100. Well, a picture is worth a thousand pictures, so here is a graph of that issue:

<img width="1024" alt="Screenshot 2023-06-06 at 16 31 57" src="https://github.com/clauseggers/Playfair/assets/868987/8540d145-400e-4953-b894-7a79073233f8">

The spreadsheet is in the directory https://github.com/clauseggers/Playfair/blob/master/sources/Settings/Playfair%202.0%20width%20axis%20mappings.numbers

**Comment by clauseggers** [comment]
Hey @davelab6 post a link to the section of the OpenType spec that says this width axis must be linear.

**Comment by clauseggers** [comment]
So @davelab6 did you mean the CSS specification? This one? https://www.w3.org/TR/css-fonts-4/#font-stretch-prop
I fail to see how that would preclude us from using a mapping such that the `100` value is placed correctly. But do enlighten me.

**Comment by davelab6** [comment]
@clauseggers please could you summarize how this was resolved and close please

**Comment by davelab6** [comment]
@clauseggers please summarize how this was addressed and close please
